subxt/src/events/event_subscription.rs

```diff
@@ -30,112 +26,16 @@ use std::{
-    let block_subscription = subscribe_to_block_headers_filling_in_gaps(
-        client,
-        last_finalized_block_number,
-        client.rpc().subscribe_finalized_blocks().await?,
-    );
-
-    Ok(EventSubscription::new(client, Box::pin(block_subscription)))
-}
-
-/// Take a subscription that returns block headers, and if any block numbers are missed out
-/// betweem the block number provided and what's returned from the subscription, we fill in
-/// the gaps and get hold of all intermediate block headers.
-///
-/// **Note:** This is exposed so that we can run integration tests on it, but otherwise
-/// should not be used directly and may break between minor releases.
-#[doc(hidden)]
-pub fn subscribe_to_block_headers_filling_in_gaps<'a, S, E, T: Config>(
-    client: &'a Client<T>,
-    mut last_block_num: Option<u64>,
-    sub: S,
-) -> impl Stream<Item = Result<T::Header, BasicError>> + Send + 'a
-where
-    S: Stream<Item = Result<T::Header, E>> + Send + 'a,
-    E: Into<BasicError> + Send + 'static,
-{
-    sub.flat_map(move |s| {
-        // Get the header, or return a stream containing just the error. Our EventSubscription
-        // stream will return `None` as soon as it hits an error like this.
-        let header = match s {
-            Ok(header) => header,
-            Err(e) => return Either::Left(stream::once(async { Err(e.into()) })),
-        };
-
-        // We want all previous details up to, but not including this current block num.
-        let end_block_num = (*header.number()).into();
-
-        // This is one after the last block we returned details for last time.
-        let start_block_num = last_block_num.map(|n| n + 1).unwrap_or(end_block_num);
-
-        // Iterate over all of the previous blocks we need headers for, ignoring the current block
-        // (which we already have the header info for):
-        let previous_headers = stream::iter(start_block_num..end_block_num)
-            .then(move |n| {
-                async move {
-                    let hash = client.rpc().block_hash(Some(n.into())).await?;
-                    let header = client.rpc().header(hash).await?;
-                    Ok::<_, BasicError>(header)
-                }
-            })
-            .filter_map(|h| async { h.transpose() });
-
-        // On the next iteration, we'll get details starting just after this end block.
-        last_block_num = Some(end_block_num);
-
-        // Return a combination of any previous headers plus the new header.
-        Either::Right(previous_headers.chain(stream::once(async { Ok(header) })))
-    })
-}
-
 /// A `jsonrpsee` Subscription. This forms a part of the `EventSubscription` type handed back
 /// in codegen from `subscribe_finalized`, and is exposed to be used in codegen.
 #[doc(hidden)]
-pub type FinalizedEventSub<'a, Header> = BoxStream<'a, Result<Header, BasicError>>;
+pub type FinalizedEventSub<Header> = BoxStream<'static, Result<Header, Error>>;
```

Comment:
what's the reason of the change? 
ease of use with less lifetime stuff?

Reply:
Yeah, I don't remember exactly but think I just wanted to simplify the lifetime stuff, and doing the odd `Arc::clone` instead made it easier to pass the client around and not have any lifetime issues moving stuff between threads and whatnot.

Reply:
(oh and to add; with that in place I guess this no longer needed to be tied to the lifetime of the client, since the client is no longer a reference)